go/vt/vttablet/tabletmanager/action_agent.go

```diff
@@ -355,6 +356,16 @@ func NewActionAgent(
 			agent.initHealthCheck()
 		}()
 	} else {
+		// optionally populate metadata records
+		if *initPopulateMetadata {
+			// we can use agent.initialTablet here because it is set by Start()
+			localMetadata := agent.getLocalMetadataValues(agent.initialTablet.Type)
+			err := mysqlctl.PopulateMetadataTables(agent.MysqlDaemon, localMetadata, topoproto.TabletDbName(agent.initialTablet))
```

Comment:
Should we add a `MysqlDaemon.Wait()` before this? Wasn't the original problem that we were trying to populate local metadata before mysqld was up?

Reply:
That's a good idea. I mistakenly thought that we can be sure that it is up at this point, but now I see that we are only initializing the vttablet side of things so we can't make that assumption.